solutions/USACO dec16-lasers-mirrors.mdx

```diff
@@ -0,0 +1,78 @@
+---
+id: usaco-671
+source: USACO Gold 2016 December
+title: Lasers and Mirrors
+author: Maggie Liu
+---
+
+[Official Analysis](http://www.usaco.org/current/data/sol_lasers_gold_dec16.html)
+
+It is optimal to use each fence post at most once, so we can use BFS to find the minimum number of fence posts needed to direct the laser to the barn. We first map a vector of fence post indices to their x and y coordinates, which we store in $\texttt{lines}$. In the queue $q$, we store the index and a boolean representing the direction of the incoming beam for each point, with true representing a horizonal incoming beam and false representing a vertical one. The vector $\texttt{dist}$ will store the number of edges of the shortest path from the laser to each point.
```

Comment:
> The vector $\texttt{dist}$ will store the number of edges of the shortest path from the laser to each point.

"The number of edges of the shortest path"?? That doesn't make sense. Shouldn't it be "the number of edges **with** the shortest path from the laser to each point"??

Also, this is optional, but you should say the "array $\texttt{dist}$", so people who use Java won't be confused...